sdk/textanalytics/Azure.AI.TextAnalytics/src/Validation.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.CompilerServices;
+using static Azure.AI.TextAnalytics.TextAnalyticsClientOptions;
+
+namespace Azure.AI.TextAnalytics
+{
+    internal static class Validation
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void SupportsOperation(string name, ServiceVersion added, ServiceVersion current)
+        {
+            if (current < added)
```

Comment:
I wonder if there's a way we can move this closer to the ServiceVersion enum definition so people updating it remember to patch up this logic?  I'm thinking about particularly screwy cases where we do things like rename `xyz_preview` to `Xyz_Preview`. I think the enum math should generally hold, but I'd love to give devs a chance to update that if they insist on doing something clever with their ServiceVersion.

Reply:
But what version added what features is invariant, hence why I didn't add it to the `ServiceVersion` originally - something along those lines did cross my mind. As for validation, that's why I added tests for all covered cases. If the tests break, the dev regressed something.

Reply:
It might not always be - especially if someone does something against guidance and different clouds support different features in the same api-version.

Reply:
> different clouds support different features in the same api-version

<img src="https://media1.giphy.com/media/XfsQPWV4nsiJ94NPRi/giphy.gif"/>

Reply:
I updated #30852 mentioning this and probably passing in some `IComparer<T>` for the service version if more complex rules are needed; however, since they are not here, I don't want to overload this implementation too much considering we may not even want to do it long term.